store/displayPreferences.ts

```diff
@@ -0,0 +1,294 @@
+import { DisplayPreferencesDto } from '@jellyfin/client-axios';
+import { MutationTree, ActionTree, GetterTree } from 'vuex';
+import nuxtConfig from '~/nuxt.config';
+
+const stringToBoolean = (value: string) => value === 'True';
+const booleanToString = (value: boolean) => (value ? 'True' : 'False');
+
+export interface DisplayPreferencesState extends DisplayPreferencesDto {
+  CustomPrefs: {
+    [key: string]: string;
+  };
+}
+
+const defaultCustomPrefs: {
+  [key: string]: string;
+} = {
+  darkMode: booleanToString(
+    nuxtConfig.vuetify?.theme?.default === 'dark' || false
+  ),
+  locale: nuxtConfig.i18n?.defaultLocale || 'en'
+};
+
+const defaultState = (): DisplayPreferencesState => ({
+  CustomPrefs: defaultCustomPrefs
+});
+
+const updateMethods: { [key: string]: (value: string) => void } = {
+  darkMode: (value: string) => {
+    window.$nuxt.$vuetify.theme.dark = stringToBoolean(value);
+  },
+  locale: (value: string) => {
+    window.$nuxt.$i18n.setLocale(value);
```

Comment:
Initially, when loading the page, it is calling this before `window.$nuxt` is defined. Probably best to check if `$nuxt` exists before trying to set the relevant mode

Reply:
Yeah that's right, I put the try catch in the `callAllCallbacks` method but that's stupid if we wanna call them one by one like in `EDIT_PREF`

Reply:
Ah ok, I missed that, that makes sense.

Reply:
No worry, I just put individual try catch in the locale and darkMode methods. This way even when we call them individually it catches the error. And this allows us to use "property methods" without a try catch.